lib/src/configuration.cc

```diff
@@ -683,6 +683,22 @@ void Configuration::validateAndNormalizeOtherSettings() {
         }
 
         auto pid_dir = fs::path(pid_file).parent_path().string();
+
+#ifdef __sun
+        // Solaris cleans up /var/run when rebooting; daemonization
+        // assumes that the PID file directory exists. If we're using
+        // the default directory, make sure it exists (PCP-297)
+        if (pid_dir == "/var/run/puppetlabs" && !fs::exists(pid_dir)) {
```

Comment:
@kylog yeah, I thought about that. But pxp-agent does not create any directories, apart from the spool ones; we followed such principle in the `Configuration` class. So, what I've tried to do here was to limit the case where we create the PID directory (Solaris platform AND default PID dir).

Also, @richardc was considering  a way to ensure the existence of the PID directory that would not involve pxp-agent directly.